src/main/java/com/flipkart/zjsonpatch/JsonPatch.java

```diff
@@ -31,10 +31,20 @@ public final class JsonPatch {
     private JsonPatch() {
     }
 
+    private static JsonNode getPatchStringAttr(JsonNode jsonNode, String attr) {
+        JsonNode child = getPatchAttr(jsonNode, attr);
+
+        if (!child.isTextual())
+            throw new InvalidJsonPatchException("Invalid JSON Patch payload (non-text '" + attr + "' field)");
```

Comment:
I wonder if we need to support or do we currently support non-text values?

Reply:
This is only for the "operation", "from" and "path" attribute which needs to be a String as per the RFC. The "value" attribute can be a struct and we are not changing anything for that.

So I would say no additional support needs to be added since these attributes need to be String to be considered valid Json patches.